.github/workflows/rust.yml

```diff
@@ -43,4 +43,36 @@ jobs:
       - name: Check CGroup
         run: systemd-run --user --scope cat /proc/self/cgroup
       - name: Run example ${{ matrix.example }}
-        run: nix develop --command systemd-run-example-${{ matrix.example }} --duration $DURATION
+        shell: nix develop --command bash -e {0}
+        run: systemd-run-example-${{ matrix.example }} --duration $DURATION 2>&1 | tee ./output.log
+      - name: Verify output
+        run: |
+          ! grep "ERROR" ./output.log || false
+          ! grep "panic" ./output.log || false
+
+          if [ "${{ matrix.example }}" = "hello_part" ]; then
+            ! grep "WARN" ./output.log || false
+            grep "Received via Sampling Port: CustomMessage" ./output.log ||
+              { printf "no custom message received"; exit 1; }
+          fi
+          if [ "${{ matrix.example }}" = "fuel_tank" ]; then
+            ! grep "WARN" ./output.log || false
+          fi
+          if [ "${{ matrix.example }}" = "ping" ]; then
+            grep "received valid response" ./output.log ||
+              { printf "no valid response received"; exit 1; }
+          fi
+          if [ "${{ matrix.example }}" = "dev_random" ]; then
+            ! grep "WARN" ./output.log || false
+            grep "got some randomness" ./output.log ||
+              { printf "missing randomness log info"; exit 1; }
+          fi
+          if [ "${{ matrix.example }}" = "ping_queue" ]; then
+            grep "Received valid response" ./output.log ||
+              { printf "no valid response received"; exit 1; }
+          fi
+          if [ "${{ matrix.example }}" = "redirect_stdio" ]; then
+            ! grep "WARN" ./output.log || false
+            grep "Terminating partition" ./output.log ||
+              { printf "partition didn't terminate as expected"; exit 1; }
+          fi
```

Comment:
Define an assert function once for more readability. Compare to https://github.com/torokmark/assert.sh/blob/main/assert.sh